extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/core/GreenbidsInvocationService.java

```diff
@@ -32,21 +32,24 @@ public GreenbidsInvocationResult createGreenbidsInvocationResult(
         final String greenbidsId = UUID.randomUUID().toString();
         final boolean isExploration = isExploration(greenbidsConfig, greenbidsId);
 
-        final BidRequest updatedBidRequest = isExploration
+        final List<Imp> updatedImps = updateImps(bidRequest, impsBiddersFilterMap);
+
+        final BidRequest updatedBidRequest = (isExploration || updatedImps.isEmpty())
```

Comment:
Also there is a potential bug with such implementation. You need to update `bidRequest` that provided in `payloadUpdate(payload)`, so you don't miss updates from other modules in same group on same stage.